plugins/offers_invreq_hook.c

```diff
@@ -258,15 +259,15 @@ static u8 *create_enctlv(const tal_t *ctx,
 			 /* in and out */
 			 struct privkey *blinding,
 			 const struct pubkey *node_id,
-			 struct short_channel_id *next_scid,
+			 const struct pubkey *next_node_id,
 			 struct tlv_encrypted_data_tlv_payment_relay *payment_relay,
 			 struct tlv_encrypted_data_tlv_payment_constraints *payment_constraints,
 			 u8 *path_secret,
 			 struct pubkey *node_alias)
 {
 	struct tlv_encrypted_data_tlv *tlv = tlv_encrypted_data_tlv_new(tmpctx);
 
-	tlv->short_channel_id = next_scid;
+	tlv->next_node_id = cast_const(struct pubkey *, next_node_id);
```

Comment:
I did not know about this `cast_const`, `ccan` is like the std that C should have